specs/20260411-002222-agent-skills-sync/contracts/walker-interface.md

```
@@ -0,0 +1,129 @@
+| - | ------- | --------------------------- | ------------------- |
+| 1 | Empty tmp dir | 0 | `[]` |
+| 2 | Missing tmp dir (not created) | 0 | `[]` |
+| 3 | Real skill `my-skill/SKILL.md` + `my-skill/README.md` | 1 | `[]` |
+| 4 | Real skill `my-skill/` with no `SKILL.md` | 0 | `[]` |
+| 5 | Real skill `my-skill/` where `SKILL.md` is a symlink to another file | 0 | `[]` |
+| 6 | Top-level symlink `my-skill -> /tmp/other/my-skill` with valid target | 0 | `[]` |
+| 7 | Top-level `.system/` directory containing a real skill | 0 | `[]` |
+| 8 | Top-level `.DS_Store` regular file | 0 | `[]` |
+| 9 | Real skill `a/SKILL.md` + another real skill `b/SKILL.md` + symlinked root `c/` | 2 | `[]` |
+| 10 | Real skill whose interior contains a symlinked helper file | 1 | `[]` (the helper is omitted silently) |
+| 11 | Real skill containing a file whose path matches `NEVER_SYNC_PATTERNS` (e.g. `./credentials.json`) | 0 (the skill is not archived) | 1 warning with prefix `never-sync inside skill: ` |
```

markdownlint-cli2 (0.22.0)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)